src/android/com/adobe/phonegap/push/PushPlugin.java

```diff
@@ -179,6 +181,20 @@ public void run() {
                     }
                 }
             });
+        } else if (SET_APPLICATION_ICON_BADGE_NUMBER.equals(action)) {
```

Comment:
Not fully no, that's what I said, not really sure how. But with content available, you could still update in background.
But for background, this could work similarly right? Not sure how to fix it otherwise.